internal/plugins/openshift/v1/init.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"regexp"
+
+	"github.com/spf13/afero"
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+)
+
+const (
+	// The current OCP release version.
+	ocpProductVersion = "4.8"
+	// The currently used version of ubi8/ubi-minimal images.
+	ubiMinimalVersion = "8.3"
+)
+
+type initSubcommand struct {
+	config config.Config
+}
+
+func (s *initSubcommand) InjectConfig(c config.Config) error {
+	s.config = c
+	return nil
+}
+
+// runInit updates a newly initialized project with OpenShift-specific configuration.
+func (s *initSubcommand) Scaffold(fs machinery.Filesystem) error {
+	if err := replaceImages(fs); err != nil {
+		return err
+	}
+
+	// Update the plugin config section with this plugin's configuration.
+	if err := s.config.EncodePluginConfig(pluginKey, Config{}); err != nil && !errors.As(err, &config.UnsupportedFieldError{}) {
+		return fmt.Errorf("error writing plugin config for %s: %v", pluginKey, err)
+	}
+
+	return nil
+}
+
+type substitution struct {
+	fromTagRE *regexp.Regexp
+	toTag     string
+}
+
+// imageSubstitutions is a map of paths to image substitutions.
+var imageSubstitutions = map[string][]substitution{
```

Comment:
nit: Do we usually do variable initialization this far down? I thought this was a function when I first read this code.  I guess it is closer to the replacement method below so not worth moving.

Reply:
It's a matter of preference, but my preference is to define globals used by few/one function/method just above or below those functions.